### PR TITLE
Clean up separation between app/database/network layers.

### PR DESCRIPTION
**Pull request by johnjohndoe**
# Description
+ Clean up separation between `app`/`database`/`network` layers for `Session` and `Meta` model classes.
  + `network` models are only used in the context of fetching data from the network and parsing it.
  + `database` models are only used in the context of loading data from the database.
  + Both `network` and `database` models are only used in `AppRepository` which acts as central junction for converting models between the layers.
  + For the app layer `AppRepository` only exposes data as `app` models.
+ Remove unneeded `:app:Highlight` model class.
+ Increase test coverage for `Session#sanitize` extension function.
+ Trim white space when parsing `speakers` string.

# Successfully tested on
with `ccc37c3` flavor, `debug` build
- :heavy_check_mark: Samsung Galaxy Tab S7 FE, SM-T733 (tablet), Android 14 (API 34)

### CODE REVIEW

**Comment by johnjohndoe** [comment]
@cketti If you have the energy to look into this as well - then you are welcome. No pressure!

**Review by cketti — DISMISSED** [review]
I think using the names `SessionAppModel`, `SessionDatabaseModel`, and `SessionNetworkModel` is a good (first) step. However, I wish you renamed the classes instead of only using import aliases.

Having different names clearly shows there's a lot of conversions going on between the different model classes. I think this could be avoided and the code would be simplified by having a single `Session` class in an API module. Then instead of calling it the "network" module, you think of it as a data provider module where the network access and source data format are just implementation details. The data provider module would parse and convert the source data to `Session` instances internally and only pass those to the app.
Similarly, the database module could use that `Session` class for communication with the app and do all necessary conversions internally.
UI code probably shouldn't use this class, but have specialized UI models that contain exactly the data that is necessary for a particular screen.

File: app/src/main/java/nerd/tuxmobil/fahrplan/congress/dataconverters/SessionExtensions.kt [hunk 1]
```diff
@@ -41,42 +42,43 @@ fun Session.toSessionDatabaseModel() = SessionDatabaseModel(
...
         changedTitle = changedTitle,
-        changedTrack = changedTrack
-)
+        changedTrack = changedTrack,
+    )
+}
 
-fun SessionDatabaseModel.toSessionAppModel(): Session {
-    return Session(
+fun SessionDatabaseModel.toSessionNetworkModel(): SessionNetworkModel {
```
[the remark this answers is not in the document]
Comment (johnjohndoe) [reply app/src/main/java/nerd/tuxmobil/fahrplan/congress/dataconverters/SessionExtensions.kt:77]: In the scope of fetching data from the network and parsing it, there is also the step where old and new data is compared to detect changes. The comparison happens before new data is written to the database. See `ScheduleChanges#computeSessionsWithChangeFlags`. I decided to deal with the same model type in this class. Therefore, the old data is loaded from the database and converted into the network model.

File: app/src/test/java/nerd/tuxmobil/fahrplan/congress/models/SessionTest.kt [hunk 2]
```diff
@@ -103,6 +103,20 @@ class SessionTest {
...
+            of(false, false, false, false, false, false, false, false, false, false, false),
+            of(true, false, false, false, false, false, false, false, false, false, true),
+            of(false, true, false, false, false, false, false, false, false, false, true),
+            of(false, false, true, false, false, false, false, false, false, false, true),
+            of(false, false, false, true, false, false, false, false, false, false, true),
+            of(false, false, false, false, true, false, false, false, false, false, true),
+            of(false, false, false, false, false, true, false, false, false, false, true),
+            of(false, false, false, false, false, false, true, false, false, false, true),
+            of(false, false, false, false, false, false, false, true, false, false, true),
+            of(false, false, false, false, false, false, false, false, true, false, true),
```
[the remark this answers is not in the document]
Comment (johnjohndoe) [reply app/src/test/java/nerd/tuxmobil/fahrplan/congress/models/SessionTest.kt:117]: Very true ... I feel guilty here. Also: nice catch of the missing case.
I moved the test cases into its own - now very readable file.
Thank you for the inspiration and the push.

**Comment by johnjohndoe** [comment]
Thank you for your feedback!

In my understanding is that the `AppRepository` is the data provider for the UI / app layer. It references the network and database layers to fulfill its purpose and to hide implementation details like the database and network models from the app layer. The goal of this pull request is to prevent the `AppRepository` from exposing anything other than the app layer model (to the best of its ability).

In its function of loading and converting data I try to keep the `AppRepository` as "uninvolved" as possible. Any screen-specific data conversion is done in the view model of that screen. - There is a temptation to move this kind of data preparation from a view model into the `AppRepository`. I think this would tell the `AppRepository` too much about the screen and is not a good separation of concerns.

I am happy to brainstorm ideas to improve the architecture. This probably works better in a synchronous conversation. Let's continue apart from this merge request.

**Comment by johnjohndoe** [comment]
@cketti Are you fine with resolving the remaining review comments?

File: app/src/main/java/nerd/tuxmobil/fahrplan/congress/dataconverters/SessionExtensions.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (johnjohndoe) [reply app/src/main/java/nerd/tuxmobil/fahrplan/congress/dataconverters/SessionExtensions.kt:77]: No further items to discuss here as agreed offline. Closing here.

File: app/src/test/java/nerd/tuxmobil/fahrplan/congress/models/SessionTest.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (johnjohndoe) [reply app/src/test/java/nerd/tuxmobil/fahrplan/congress/models/SessionTest.kt]: No further items to discuss here as agreed offline. Closing here.

**Pull request merged**